frontend/packages/dev-console/src/components/import/GitImportForm.tsx

```diff
@@ -26,28 +27,42 @@ const GitImportForm: React.FC<FormikProps<FormikValues> & GitImportFormProps> =
   status,
   isSubmitting,
   dirty,
-}) => (
-  <Form onReset={handleReset} onSubmit={handleSubmit}>
-    <div className="co-m-pane__form">
-      <GitSection />
-      <AppSection project={values.project} />
-      <BuilderSection image={values.image} builderImages={builderImages} />
-      <ExpandCollapse>
-        <RouteSection route={values.route} />
-        <BuildConfigSection namespace={values.project.name} />
-        <DeploymentConfigSection namespace={values.project.name} />
-        <ScalingSection />
-        <LabelSection />
-      </ExpandCollapse>
-    </div>
-    <br />
-    <ButtonBar errorMessage={status && status.submitError} inProgress={isSubmitting}>
-      <Button disabled={!dirty || !_.isEmpty(errors)} type="submit" bsStyle="primary">
-        Create
-      </Button>
-      <Button type="reset">Cancel</Button>
-    </ButtonBar>
-  </Form>
-);
+}) => {
+  const [isExpanded, toggleExpandCollapse] = React.useState(false);
+
+  const ontoggle: any = (event) => {
+    // TODO: This can be removed when https://github.com/patternfly/patternfly-react/issues/2339 is fixed
+    event.preventDefault();
+    toggleExpandCollapse(!isExpanded);
+  };
+
+  return (
+    <Form onReset={handleReset} onSubmit={handleSubmit}>
+      <div className="co-m-pane__form">
+        <GitSection />
+        <AppSection project={values.project} />
+        <BuilderSection image={values.image} builderImages={builderImages} />
+        <Expandable
```

Comment:
You can create this as a separate `Formik` field, that way you don't need to add these details here. All of it will be handled by the `Formik` field. You can get the `isExpanded` value from the state using `useFormikContext`/`useField`, also `onToggle` can be called with a function that uses `setFieldValue` to update the state.